repo/common/common.go

```diff
@@ -2,26 +2,50 @@ package common
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 )
 
+// Find dynamic map key names passed  as Parent["foo"] notation
+var bracketsRe = regexp.MustCompile(`\[([^\[\]]*)\]`)
+
+// Normalization for supporting arbitrary dynamic keys with dots:
+// Gateway.PublicGateways["gw.example.com"].UseSubdomains
+// Pinning.RemoteServices["pins.example.org"].Policies.MFS.Enable
+func keyToLookupData(key string) (normalizedKey string, dynamicKeys map[string]string) {
+	bracketedKeys := bracketsRe.FindAllString(key, -1)
+	dynamicKeys = make(map[string]string, len(bracketedKeys))
+	normalizedKey = key
+	for i, mapKeySegment := range bracketedKeys {
+		mapKey := strings.TrimLeft(mapKeySegment, "[\"")
+		mapKey = strings.TrimRight(mapKey, "\"]")
```

Comment:
These remove all instances. While I'd certainly hope that nobody does anything like `field["["foobar"]` I'd think that should cause some kind of error instead of getting silently fixed.

Reply:
Ah, nice catch. Changed to Prefix/Suffix in  732adebaf